core/blocks/src/categories.rs

```diff
@@ -1,184 +1,75 @@
 use crate::{BlockId, BlockKind};
+use feather_definitions::SimplifiedBlockKind;
+
+pub enum PlacementType {
+    TargetedFace,
+    PlayerDirection,
+    PlayerDirectionRightAngle,
+}
+
+pub enum SupportType {
+    OnSolid,
+    OnDesertBlocks,
+    OnDirtBlocks,
+    OnFarmland,
+    OnSoulSand,
+    OnWater,
+
+    FacingSolid,
+    FacingJungleWood,
+
+    OnOrFacingSolid,
+
+    CactusLike,
+    ChorusFlowerLike,
+    ChorusPlantLike,
+    MushroomLike,
+    SnowLike,
+    SugarCaneLike,
+    TripwireHookLike,
+    VineLike,
+}
 
 impl BlockId {
+    #[inline(always)]
```

Comment:
What's the reason for the inlining here? I understand it might yield performances in some circumstances, but in general I'd prefer to leave it to the optimizer to decide on inlining. A compromise would be just `#[inline]`, which allows for cross-crate inlining but doesn't force the optimizer to do so. `#[inline(always)]` is generally discouraged in Rust, from what I can tell.

Reply:
Hmm, yes, I wasn't that aware of the distinction between `#[inline]` and `#[inline(always)]`. `#[inline]` definitely makes more sense in this case then. Though I wouldn't remove the hint completely, because I suspect inlining these methods can make a lot of sense, especially when only forwarding to another function or matching on the return value (SupportType, PlacementType) in the caller.